### DISCUSSION TITLE
Импорт данных из эксель в простую таблицу

### DISCUSSION THREAD

**Discussion by horyachka** [totumonline/totum-ru-issues-and-discussions#32] [Как это сделать?] [answered]
Добрый день.

Как реализовать возможность импорта данных в простую таблицу для пользователей, которые ежедневно вносят данные, но хотят вносить данные массивом из экселя.

#### Пару слов о ОСНОВНОЙ таблице в которую импортируются данные:

+ Таблицу заполняют два пользователя: отправитель и получатель.

+ Получатель и отправитель видят все ячейки в строке, но могут заполнять только свои ячейки.

+ **Некоторые поля, которые нужно импортировать, являются Селектами и пользователи хотят загружать не их значения, а отображения и нам при загрузке надо автоматически превращать в значения и не давать загрузить ошибочные!**

+ Кроме этого на основе данных этой таблицы созданы отчеты:
  + остатки тары;
  + движение тары за период;
  + и другие.

Спасибо!

**UPD:** Вопрос упрощен by @ttm-alexeypolunin, что бы на него была возможность ответить здесь.

**Comment by ttm-alexeypolunin** [comment, marked as answer]
### [Видео на YouTube —>](https://youtu.be/V7W4w5I1TKQ)

В качестве ответа — смотреть видео.

#### Некоторые коды, которые объясняются в видео:

Код кнопки Convert:

```
= : linkToDataTable(table: 's4_tmp_2'; title: "Converted"; width: 700; height: "80vh"; data: $rowList; params: $param;  refresh: true)

rowList: listReplace(list: $selrowlist; action: "field_3" = $val; value: "value")
	val: if(condition: $selval = ""; then: $#value[field_3]; else: $selval)
		selval: select(table: 's4_select_table'; field: 'id'; where: 'title' = $#value[field_3])

	selrowlist: selectRowList(table: $#ntn; fields: "*ALL*")
	
	param: rowCreate(field: "f_hash" = $#nth)

```

> Инструкция `"*ALL*"` в `selectRowList(table: $#ntn; fields: "*ALL*")` поддерживается только со 2 версии Тотума. В 1 надо явно перечислить поля в виде списка.

Код кнопки Insert:

```

a1= : insertList(table: 's4_simple'; field: 'field_1' = $list[[field_1]]; field: 'field_2' = $list[[field_2]]; field: 'select_field' = $list[[field_3]]; log: true)

~list: selectRowList(table: $#ntn; field: 'field_1'; field: 'field_2'; field: 'field_3')

a2=: deleteList(table: 's4_tmp_csv'; hash: #f_hash)

```

**Reply by lutikoff** [reply]
Чего то в примере не хватает, сделал в тесте всё тоже самое но по кнопке конверт получаю пустые строки.
![Снимок экрана 2024-04-17 в 12 20 11](https://github.com/totumonline/totum-ru-issues-and-discussions/assets/167204752/78a895cc-3e06-4f21-8e97-49b586f2425e)

**Comment by horyachka** [comment]
В моем проекте все заработало﻿
Чего то в примере не хватает, сделал в тесте всё тоже самое но по кнопке конверт получаю пустые строки.
2024-04-17.12.20.11.png (view on web)

**Reply by lutikoff** [reply]
Да, действительно была пара ошибок, всё заработало.
А вот такой вопрос, поле select field доступен выбор множественного значения, и например значения перечислены или в новых столбцах или одной ячейке и например через точку с запятой?
